src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -87,7 +87,10 @@ internal static class ProjectRestoreInfoBuilder
             return targetFrameworks.Any()
                 ? new ProjectRestoreInfo
                 {
-                    BaseIntermediatePath = baseIntermediatePath,
+                    // NOTE: We pass MSBuildProjectExtensionsPath as BaseIntermediatePath, because people often set
+                    // BaseIntermediatePath too late.  See https://github.com/dotnet/project-system/issues/3466 for
+                    // details.
```

Comment:
Note that the MSBuild property is `BaseIntermediateOutputPath`, which is not the same as the property name: `BaseIntermediatePath`.  I would reword the comment similar to this:

NOTE: We pass MSBuildProjectExtensionsPath as BaseIntermediatePath instead of using BaseIntermediateOutputPath.  This is because NuGet switched from using BaseIntermediateOutputPath to MSBuildProjectExtensionsPath, since the value of BaseIntermediateOutputPath is often set too late (after *.g.props files would need to have been imported from it).  Instead of modifying the IVsProjectRestoreInfo interface or introducing something like IVsProjectRestoreInfo with an MSBuildProjectExtensionsPath property, we opted to leave the interface the same but change the meaning of its BaseIntermediatePath proprtey.  See https://github.com/dotnet/project-system/issues/3466 for for details.

Reply:
Done thanks!